python/pyarrow/io.pxi

```diff
@@ -1327,6 +1401,20 @@ def allocate_buffer(int64_t size, MemoryPool memory_pool=None,
 
 
 cdef class BufferOutputStream(NativeFile):
+    """
+    Examples
```

Comment:
Can you add a summary first line of the docstring? (before the "Examples" section)